File: XiaomiParts/src/com/xiaomi/settings/BootCompletedReceiver.java
/*
 * Copyright (C) 2023-2024 Paranoid Android
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.xiaomi.settings;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.UserHandle;
import android.util.Log;

import com.xiaomi.settings.battery.ChargingLimitService;
import com.xiaomi.settings.display.ColorModeService;

public class BootCompletedReceiver extends BroadcastReceiver {
    private static final String TAG = "XiaomiParts";
    private static final boolean DEBUG = true;

    @Override
    public void onReceive(final Context context, Intent intent) {
        if (DEBUG) Log.i(TAG, "Received intent: " + intent.getAction());
        switch (intent.getAction()) {
            case Intent.ACTION_LOCKED_BOOT_COMPLETED:
                onLockedBootCompleted(context);
                break;
            case Intent.ACTION_BOOT_COMPLETED:
                onBootCompleted(context);
                break;
        }
    }

    private static void onLockedBootCompleted(Context context) {
        // Display
        context.startServiceAsUser(new Intent(context, ColorModeService.class),
                UserHandle.CURRENT);
        // Battery
        context.startServiceAsUser(new Intent(context, ChargingLimitService.class),
                UserHandle.CURRENT);
    }

    private static void onBootCompleted(Context context) {
    }
}

File: XiaomiParts/src/com/xiaomi/settings/BootCompletedReceiver.kt
/*
 * Copyright (C) 2023-2024 Paranoid Android
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.xiaomi.settings

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.UserHandle
import android.util.Log

import com.xiaomi.settings.battery.ChargingLimitService
import com.xiaomi.settings.display.ColorModeService

class BootCompletedReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context?, intent: Intent?) {
        val action = intent?.action ?: return
        val ctx = context ?: return

        if (DEBUG) Log.i(TAG, "Received intent: $action")

        when (action) {
            Intent.ACTION_LOCKED_BOOT_COMPLETED -> onLockedBootCompleted(ctx)
            Intent.ACTION_BOOT_COMPLETED -> onBootCompleted(ctx)
        }
    }

    private fun onLockedBootCompleted(context: Context) {
        // Display
        context.startServiceAsUser(
            Intent(context, ColorModeService::class.java),
            UserHandle.CURRENT
        )
        // Battery
        context.startServiceAsUser(
            Intent(context, ChargingLimitService::class.java),
            UserHandle.CURRENT
        )
    }

    private fun onBootCompleted(context: Context) {
    }

    companion object {
        private const val TAG = "XiaomiParts"
        private const val DEBUG = true
    }
}
